pkg/workspace_repo/workspace.go

```diff
@@ -242,6 +267,35 @@ func (s *TFEWorkspace) WaitWithOptions(waitOptions options.WaitOptions) error {
 	return nil
 }
 
+func (s *TFEWorkspace) streamLogs(ctx context.Context, logs io.Reader) {
+	logrus.Info("...streaming logs...")
+	reader := bufio.NewReaderSize(logs, 64*1024)
```

Comment:
how about `bufio.Scanner`? I think it might give you a better level of abstraction